src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java

```diff
@@ -39,14 +37,38 @@ public void onClick(PlayerInteractEvent event) {
             return;
         }
 
-        event.getClickedBlock().setType(Material.AIR);
-        if (itemInHand.getAmount() > 1) {
-            itemInHand.setAmount(itemInHand.getAmount() - 1);
-            player.getInventory().addItem(new ItemStack(Material.LAVA_BUCKET)).values().forEach(itemStack ->
-                    player.getWorld().dropItem(player.getLocation(), itemStack)
-            );
-        } else {
-            itemInHand.setType(Material.LAVA_BUCKET);
+        if(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
+                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && event.getClickedBlock().getType().equals(Material.OBSIDIAN)
+                && itemInHand.getType().equals(Material.BUCKET)) {
+
+            event.getClickedBlock().setType(Material.AIR);
+            if (itemInHand.getAmount() > 1) {
+                itemInHand.setAmount(itemInHand.getAmount() - 1);
+                player.getInventory().addItem(new ItemStack(Material.LAVA_BUCKET)).values().forEach(itemStack ->
+                        player.getWorld().dropItem(player.getLocation(), itemStack)
+                );
+            } else {
+                itemInHand.setType(Material.LAVA_BUCKET);
+            }
+            event.setCancelled(true);
+        }
+
+        if(IridiumSkyblock.getInstance().getConfiguration().endPortalPick
+                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
```

Comment:
would left clicking make more scene? Maybe only if they shift + left click aswell?

Reply:
left click makes more sense in the context of the game, but I think its easier to explain to players because it works in the same way as obsidianBucket